### PR TITLE
Remove random_numbers dependency

### PR DESCRIPTION
**Pull request by sjahr**
### Description
Depends on https://github.com/ros-planning/geometric_shapes/pull/223

This PR aims to remove the random number dependency from MoveIt2. To do so, the cpp std library `<random>` is used directly. ~~Instead of the RandomNumberGenerator, `std::mt19937` is used in combination with two simple utility functions defined in the newly added header `moveit_core/utils/include/moveit/utils/random_number_utils.hpp`~~

Instead of the random_numbers::RandomNumberGenerator, a simpler class in implemented seeded like  recommanded in the review in `moveit_core/utils/include/moveit/utils/random_number_utils.hpp

Generating random quaternions can be done with the constructor from Eigen itself. For example like this:
```
Eigen::Quaterniond rand_quaternion = Eigen::Quaterniond::UnitRandom();
```

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_core/constraint_samplers/include/moveit/constraint_samplers/default_constraint_samplers.h [hunk 1]
```diff
@@ -304,7 +308,11 @@ class IKConstraintSampler : public ConstraintSampler
    *
    */
   IKConstraintSampler(const planning_scene::PlanningSceneConstPtr& scene, const std::string& group_name)
-    : ConstraintSampler(scene, group_name)
+    : ConstraintSampler(scene, group_name) , random_number_generator_([]() {
+    std::random_device r;
+    std::seed_seq const seed{ r(), r(), r(), r(), r(), r(), r(), r() };
```
Comment (tylerjw) [inline moveit_core/constraint_samplers/include/moveit/constraint_samplers/default_constraint_samplers.h:313]: @ChrisThrasher do you by chance have a link to the snipped of code we saw at the meetup last week for this?

**Review by tylerjw — COMMENTED** [review]
Last week a few of us went to a C++ meetup where the usage of <random> was the topic and there was some really interesting and useful guidence there. I'm sad I can't seem to find any links to a writeup right now. Maybe @ChrisThrasher knows of a link to something or has advice here?

File: moveit_core/constraint_samplers/include/moveit/constraint_samplers/default_constraint_samplers.h (hunk 1 above)
Comment (ChrisThrasher) [reply moveit_core/constraint_samplers/include/moveit/constraint_samplers/default_constraint_samplers.h:313]:
```cpp
std::array<int, std::mt19937::state_size> seed_data;
std::random_device rd;
std::generate_n(std::data(seed_data), std::size(seed_data), std::ref(rd));
std::seed_seq seq(std::begin(seed_data), std::end(seed_data));
std::mt19937 rng(seq);
```

Comment (sjahr) [reply moveit_core/constraint_samplers/include/moveit/constraint_samplers/default_constraint_samplers.h:313]: Thanks!

File: moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common_panda.h [hunk 2]
```diff
@@ -266,19 +268,20 @@ TYPED_TEST_P(DistanceCheckPandaTest, DistanceSingle)
...
+    shapes::ShapeConstPtr shape =
+        std::make_shared<const shapes::Cylinder>(uniform<double>(mt, 0.0, 1.0), uniform<double>(mt, 0.0, 1.0));
     Eigen::Isometry3d pose{ Eigen::Isometry3d::Identity() };
     pose.translation() =
-        Eigen::Vector3d(rng.uniformReal(0.1, 2.0), rng.uniformReal(0.1, 2.0), rng.uniformReal(1.2, 1.7));
-    double quat[4];
-    rng.quaternion(quat);
-    pose.linear() = Eigen::Quaterniond(quat[0], quat[1], quat[2], quat[3]).toRotationMatrix();
+        Eigen::Vector3d(uniform<double>(mt, 0.1, 0.1), uniform<double>(mt, 0.1, 2.0), uniform<double>(mt, 1.2, 1.7));
+    Eigen::Quaterniond rand_quaternion = Eigen::Quaterniond::UnitRandom();
```
[the remark this answers is not in the document]
Comment (henningkayser) [reply moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common_panda.h:283]: While using rand is really not ideal, I think it should work for our use case here. For improving performance, and making randomness more consistent and easy to use, we could consider using something like https://github.com/bab2min/EigenRand. Maybe there are better options out there.

[the remark this answers is not in the document]
Comment (sjahr) [reply moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common_panda.h:283]: Ok, thanks for the feedback. I'll take a look at it and investigate a little whether or not there are other more suitable libraries available

**Comment by tylerjw** [comment]
Closing this as we are now using rng from RSL.

**Pull request closed without merging**